File: printf/tools.c
/* ************************************************************************** */
/*                                                          LE - /            */
/*                                                              /             */
/*   tools.c                                          .::    .:/ .      .::   */
/*                                                 +:+:+   +:    +:  +:+:+    */
/*   By: loiberti <marvin@le-101.fr>                +:+   +:    +:    +:+     */
/*                                                 #+#   #+    #+    #+#      */
/*   Created: 2018/11/16 21:06:00 by loiberti     #+#   ##    ##    #+#       */
/*   Updated: 2018/12/10 20:02:09 by loiberti    ###    #+. /#+    ###.fr     */
/*                                                         /                  */
/*                                                        /                   */
/* ************************************************************************** */

#include "printf.h"


char	*string_null(char *str)
{
	if (!str)
		return (ft_strdup("(null)"));
	return (str);
}

int		is_hexa(char *str)
{
	int	i;

	i = -1;
	while (str[++i])
		if ((str[i] >= 'a' && str[i] <= 'f') \
										|| (str[i] >= 'A' && str[i] <= 'F'))
			return (1);
	return (0);
}

File: CPV2/tools.py
# **************************************************************************** #
#                                                           LE - /             #
#                                                               /              #
#    tools.py                                         .::    .:/ .      .::    #
#                                                  +:+:+   +:    +:  +:+:+     #
#    By: loiberti <marvin@le-101.fr>                +:+   +:    +:    +:+      #
#                                                  #+#   #+    #+    #+#       #
#    Created: 2019/11/10 20:01:11 by loiberti     #+#   ##    ##    #+#        #
#    Updated: 2019/11/10 21:47:03 by loiberti    ###    #+. /#+    ###.fr      #
#                                                          /                   #
#                                                         /                    #
# **************************************************************************** #

def roundornot(nb):
    nb = float(nb)
    return int(nb) if nb.is_integer() else round(nb, 3)

def do_op(v1, v2, op):
    if op == "+":
        return round(float(v1) + float(v2), 3)
    elif op == "-":
        return round(float(v1) - float(v2), 3)
    elif op == "*":
        return round(float(v1) * float(v2), 3)
    elif op == "/":
        return round(float(v1) / float(v2), 3)
    elif op == "%":
        return round(float(v1) % float(v2), 3)
    elif op == "^":
        return round(float(v1) ** v2, 3)
